### PR TITLE
Improve xsystem and xasset-registrar 

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

### CODE REVIEW

File: xpallets/assets/src/lib.rs [hunk 1]
```diff
@@ -249,7 +250,7 @@ impl<T: Trait> Module<T> {
         asset_id: AssetId,
         restrictions: AssetRestrictions,
     ) -> DispatchResult {
-        xpallet_assets_registrar::Module::<T>::ensure_asset_exists(&asset_id)?;
+        xpallet_assets_registrar::Module::<T>::ensure_asset_is_exists(&asset_id)?;
```
Comment (liuchengxu) [inline xpallets/assets/src/lib.rs]: `is_exists` is incorrect in grammer. Please revert this change.

File: xpallets/system/src/lib.rs [hunk 2]
```diff
@@ -2,113 +2,139 @@
...
             }
             Ok(())
         }
 
+        /// Modify the blocked status of the given account id.
+        ///
+        /// This is a root-only operation.
         #[weight = 0]
-        fn modify_blocked_list(origin, who: <T::Lookup as StaticLookup>::Source, block: bool) -> DispatchResult {
+        fn modify_blocked_list(origin, who: <T::Lookup as StaticLookup>::Source, is_block: bool) -> DispatchResult {
```
Comment (liuchengxu) [inline xpallets/system/src/lib.rs]:
```suggestion
        fn modify_blocked_list(origin, who: <T::Lookup as StaticLookup>::Source, should_block: bool) -> DispatchResult {
```

File: xpallets/system/Cargo.toml [hunk 3]
```diff
@@ -28,14 +27,13 @@ std = [
...
     "frame-system/std",
 
     "xpallet-protocol/std",
 ]
-runtime-benchmarks = [
-    "frame-benchmarking",
-]
+#runtime-benchmarks = [
+#    "frame-benchmarking",
+#]
```
Comment (liuchengxu) [inline xpallets/system/Cargo.toml]:
```suggestion
```

File: xpallets/assets-registrar/src/lib.rs [hunk 4]
```diff
@@ -310,7 +212,8 @@ decl_module! {
             desc: Option<Desc>
         ) -> DispatchResult {
             ensure_root(origin)?;
-            let mut info = Self::asset_info_of(&id).ok_or(Error::<T>::AssetDoesNotExist)?;
+
+            let mut info = Self::asset_info_of(&id).ok_or(Error::<T>::AssetIsNotExist)?;
```
Comment (liuchengxu) [inline xpallets/assets-registrar/src/lib.rs]: `AssetIsNotExist` is incorrect in grammer.

File: xpallets/assets-registrar/src/lib.rs [hunk 5]
```diff
@@ -10,196 +10,88 @@
 
 #[cfg(any(feature = "runtime-benchmarks", test))]
 mod benchmarking;
+mod default_weights;
```
Comment (liuchengxu) [inline xpallets/assets-registrar/src/lib.rs]: What's the reason for this change?

File: xpallets/assets-registrar/src/lib.rs [hunk 6]
```diff
@@ -271,36 +165,44 @@ decl_module! {
...
         /// `RegistrarHandler::on_register()` will be triggered again during the recover process.
+        ///
+        /// This is a root-only operation.
         #[weight = T::WeightInfo::recover()]
         pub fn recover(origin, #[compact] id: AssetId, has_mining_rights: bool) -> DispatchResult {
             ensure_root(origin)?;
+
             ensure!(Self::asset_exists(&id), Error::<T>::AssetDoesNotExist);
-            ensure!(!Self::is_valid_asset(&id), Error::<T>::AssetAlreadyValid);
+            ensure!(!Self::asset_is_valid(&id), Error::<T>::AssetAlreadyValid);
```
Comment (liuchengxu) [inline xpallets/assets-registrar/src/lib.rs]: Good reason for this change?

File: xpallets/assets-registrar/src/lib.rs (hunk 5 above)
Comment (koushiro) [reply xpallets/assets-registrar/src/lib.rs]: I just want to keep consistent with the naming of substrate frame(pallet)

File: xpallets/assets-registrar/src/lib.rs (hunk 6 above)
Comment (koushiro) [reply xpallets/assets-registrar/src/lib.rs]: just want to keep consistent with `ensure_asset_is_valid`

File: xpallets/assets-registrar/src/lib.rs (hunk 5 above)
Comment (liuchengxu) [reply xpallets/assets-registrar/src/lib.rs]: Substrate is intended to be a general framework/library, but ChainX is a project with specific features based on Substrate more like Polkadot. In my opinion, `default_weights` in Substrate is to let the pallet users don't need to run the benchmarks on their own and can use the default benchmarking results directly, we don't need to follow this way.

Comment (liuchengxu) [reply xpallets/assets-registrar/src/lib.rs]: Furthermore, splitting the trait defintion and default implementation is not good for the future auto benchmarking script.

File: xpallets/assets-registrar/src/lib.rs (hunk 6 above)
Comment (liuchengxu) [reply xpallets/assets-registrar/src/lib.rs]: That's not quite sound.

Comment (koushiro) [reply xpallets/assets-registrar/src/lib.rs]: I think this reason is enough.
If I rename the function `asset_exists` to `exists_asset` but keep the name of `ensure_asset_exists`. Isn't that weird?

Comment (liuchengxu) [reply xpallets/assets-registrar/src/lib.rs]: Just keep `is_valid_asset()`. From the point of naming an API, `is_valid_asset(asset_id)` is more common than `asset_is_valid(asset_id)` which is a bit of tedious, see https://doc.rust-lang.org/std/path/fn.is_separator.html .

>If I rename the function asset_exists to exists_asset but keep the name of ensure_asset_exists. Isn't that weird?

`asset exists` is correct with respect to the English grammer, but `exists asset` is wrong.

**Review by liuchengxu — APPROVED** [review]
